numpy/core/_add_newdocs.py

```diff
@@ -7969,62 +7969,198 @@ def luf(lamdaexpr, *args, **kwargs):
+    ('uintp', 'Unsigned integer large enough to fit pointer, compatible with C ``uintptr_t``. Alias: uint0'),
+    ])
+
+float_aliases = type_aliases('numpy.core.numerictypes', [
+    ('float16', '16-bit-precision floating-point number type: sign bit, 5 bits exponent, 10 bits mantissa'),
+    ('float32', '32-bit-precision floating-point number type: sign bit, 8 bits exponent, 23 bits mantissa'),
+    ('float64', '64-bit precision floating-point number type: sign bit, 11 bits exponent, 52 bits mantissa'),
+    ('float96', '96-bit extended-precision floating-point number type'),
+    ('float128', '128-bit extended-precision floating-point number type'),
+    ])
+
+complex_aliases = type_aliases('numpy.core.numerictypes', [
+    ('complex64', 'Complex number type composed of 2 32-bit-precision floating-point numbers'),
+    ('complex128', 'Complex number type composed of 2 64-bit-precision floating-point numbers'),
+    ('complex192', 'Complex number type composed of 2 96-bit extended-precision floating-point numbers'),
+    ('complex256', 'Complex number type composed of 2 128-bit extended-precision floating-point numbers'),
+    ])
+
+def extra_alias_doc(place, obj, possible_aliases):
+    o = get_type(place, obj)
+    return ''.join("Alias *on this platform*: {}: {}.\n".format(alias, doc)
+                   for (alias_type, alias, doc) in possible_aliases if alias_type is o)
 
-add_newdoc('numpy.core.numerictypes', 'complex64',
+add_newdoc('numpy.core.numerictypes', 'bool_',
     """
-    Complex number type composed of two 32 bit floats. Character code: 'F'.
+    Boolean type (True or False), stored as a byte.
+    Character code: ``'?'``.
+    Alias: ``bool8``.
 
     """)
 
-add_newdoc('numpy.core.numerictypes', 'complex128',
+add_newdoc('numpy.core.numerictypes', 'byte',
     """
-    Complex number type composed of two 64 bit floats. Character code: 'D'.
-    Python complex compatible.
+    Signed integer type, compatible with C ``char``.
+    Character code: ``'b'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'byte', integer_aliases)))
 
-    """)
+add_newdoc('numpy.core.numerictypes', 'short',
+    """
+    Signed integer type, compatible with C ``short``.
+    Character code: ``'h'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'short', integer_aliases)))
 
-add_newdoc('numpy.core.numerictypes', 'complex256',
+add_newdoc('numpy.core.numerictypes', 'intc',
     """
-    Complex number type composed of two 128-bit floats. Character code: 'G'.
+    Signed integer type, compatible with C ``int``.
+    Character code: ``'i'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'intc', integer_aliases)))
 
-    """)
+add_newdoc('numpy.core.numerictypes', 'int_',
+    """
+    Signed integer type, compatible with Python `int` anc C ``long``.
+    Character code: ``'l'``.
+    {}
+    """.format(extra_alias_doc('numpy.core.numerictypes', 'int_', integer_aliases)))
```

Comment:
Wrapping the format string in `textwrap.dedent` would solve your indentation problem

Reply:
Apparently indenting everything equally also does the job (guessing Python's help does the `dedent`, then), see fresh commit `8a53e9d`. (And that's how it already was before this PR as well.)

Reply:
`add_newdocs` does a dedent internally